File: leetcode/0009. Palindrome Number/Solution.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        List<Integer> list = new ArrayList<Integer>();
        while(x > 0) {
            list.add(x % 10);
            x /= 10;
        }
        int digits = list.size();
        for (int i = 0; i < digits / 2; i++) {
            if (!list.get(i).equals(list.get(digits - 1 - i))) {
                return false;
            }
        }
        return true;
    }
}

File: leetcode/0009. Palindrome Number/Solution.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x == 0:
            return True
        if x < 0 or x % 10 == 0:
            return False

        # Reverse x.
        copy = x
        reverse = 0

        while copy != 0:
            last_digit = copy % 10
            copy /= 10
            reverse =  reverse * 10 + last_digit

        # Check if the reversed x and x are the same.
        if reverse == x:
            return True
        else:
            return False
        
